test/Tests/src/Data/Text/Regex_Spec.enso

```diff
@@ -33,6 +33,11 @@ spec =
         Test.specify "should disallow empty patterns in `compile`" <|
             Regex.compile "" . should_fail_with Illegal_Argument
 
+        Test.specify "passing a non-string should fail with a type error" <|
+            Test.expect_panic_with (Regex.compile 12) Type_Error
+            p = Regex.compile "[a-z]"
+            Test.expect_panic_with (Regex.compile p) Type_Error
```

Comment:
Given this behaviour (that I think is good), do we still need the `parseRegexPattern`?

I assume it should not be allowed to pass an already-compiled regex to `compile_regex`, we should use James's `recompile` instead - e.g. to update case sensitivity.

Reply:
My understanding is that parseRegexPattern is for caching compiled regexes, so if you compile the same one multiple times, it uses the cache. This test is for passing a non-string to Regex.compile.

Reply:
Ah sorry! I read that wrong apparently.